### PR TITLE
[12.0][MIG] stock_available: Migration to v12.0

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa

### CODE REVIEW

File: stock_available/README.rst [hunk 1]
```diff
@@ -33,44 +57,50 @@ This module adds a field named `Available for sale` on the Product form.
...
 Contributors
-------------
+~~~~~~~~~~~~
 
-* Lionel Sausin (Numérigraphe) <ls@numerigraphe.com>
-* Sodexis <sodexis@sodexis.com>
-* Factor Libre - Kiko Peiro <francisco.peiro@factorlibre.com>
+* `Numérigraphe <https://www.numerigraphe.com>`_:
+    * Lionel Sausin <ls@numerigraphe.com>
+* `Sodexis <https://www.Sodexis.com>`_:
```
Comment (pedrobaeza) [inline stock_available/README.rst:86]: You must leave an empty line for proper RST syntax

File: stock_available/views/product_product_view.xml [hunk 2]
```diff
@@ -14,20 +14,24 @@
                 <button type="action" name="%(stock.product_open_quants)d"
                     attrs="{'invisible':[('type', 'not in', ['product','consu'])]}"
                     class="oe_stat_button" icon="fa-building-o">
-                    <div class="o_form_field o_stat_info">
-                        <field name="immediately_usable_qty"
-                            widget="statinfo" nolabel="1" />
-                        <span class="o_stat_text">Available</span>
-                    </div>
+                        <div class="o_field_widget o_stat_info">
+                            <span class="o_stat_value" widget="statinfo">
```
Comment (pedrobaeza) [inline stock_available/views/product_product_view.xml:18]: Good change mimicking the style of v12 stock on hand and company smart-buttons!

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done!!

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@sergio-teruel I'm seeing that this module is already migrated in #540, so we have a conflict. Please always check first before migrating a module if there's another one. The other one seems to have an interesting change from 10.0, and yours have about the style on the smart-buttons. I'm going to ask in the other PR if they are interested in continuing it.

**Comment by pedrobaeza** [comment]
@sergio-teruel please include missing v10 commits that are on the other PR. Or they are here already?

**Comment by sergio-teruel** [comment]
@pedrobaeza Yes, they were already included.

**Comment by pedrobaeza** [comment]
OK, great

**Review by pedrobaeza — APPROVED** [review]

**Review by victor-ditay — APPROVED** [review]

File: stock_available/models/product_template.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# Copyright 2014 Numérigraphe
+# Copyright 2016 Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api
+from odoo.addons import decimal_precision as dp
+import operator as py_operator
+
+OPERATORS = {
```
Comment (Tardo) [inline stock_available/models/product_template.py]: This is used more than once... perhaps can be good export it to a separated .py ? for example "utils.py"

**Review by Tardo — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
@Tardo Changes done!!

File: stock_available/models/product_template.py (hunk 3 above)
Comment (rousseldenis) [inline stock_available/models/product_template.py]: I would have said that it is included in Odoo, so:

```suggestion
from odoo.addons.stock.models.product import OPERATORS
```

File: stock_available/models/product_product.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# Copyright 2014 Numérigraphe
+# Copyright 2016 Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+from .product_template import OPERATORS
```
Comment (rousseldenis) [inline stock_available/models/product_product.py]: from odoo.addons.stock.models.product import OPERATORS

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Just a change

File: stock_available/models/product_product.py (hunk 4 above)
Comment (sergio-teruel) [reply stock_available/models/product_product.py]: thanks @rousseldenis

**Comment by pedrobaeza** [comment]
Please check Travis and runbot status

**Comment by sergio-teruel** [comment]
Travis Ok

**Review by rousseldenis — APPROVED** [review]
Code review. Could you squash ?

**Review by florian-dacosta — APPROVED** [review]
Tests + code review

**Comment by pedrobaeza** [comment]
@sergio-teruel next time you edit from the GitHub interface, what you can do is to change commit message to comply with "!fixup <previous commit name>", and this way, OCA Bot will auto-squash it. Are you squashing now?

**Review by Tardo — APPROVED** [review]

**Pull request merged**